python/pytest/pytest.bzl

```diff
@@ -0,0 +1,27 @@
+"""Support for Pytest tests."""
+
+load("@rules_python//python:defs.bzl", "py_test")
+
+def pytest_test(name, srcs = [], deps = [], args = [], **kwargs):
+    pytest_args = [
+        # Disable Pytest's own caching.
+        "-p",
+        "no:cacheprovider",
+        # Generate the XML summary where Bazel expects it.
+        "--junit-xml=$$XML_OUTPUT_FILE",
+        # Print the captured output of all tests, so Bazel's --test_output works as expected.
+        "-rA",
+        # Avoid collecting tests automatically; use what's in `srcs`.
+        "--ignore=.",
+    ] + [
+        "$(location %s)" % src for src in srcs
+    ] + args
+
+    py_test(
+        name = name,
+        srcs = srcs + ["//python/pytest:pytest_runner"],
```

Comment:
Unfortunately, without that we can't use it in `main`:
```
ERROR: example/python/BUILD:33:12: in srcs attribute of py_test rule //python:hello_pytest_test: could not find 'pytest_runner.py' as specified by 'main' attribute. Since this rule was created by the macro 'pytest_test', the error might have been caused by the macro implementation
```
And when it's only in `srcs`, we aren't getting its transitive dependencies.